inference-engine/src/auto_plugin/auto_plugin.hpp

```diff
@@ -72,6 +72,9 @@ class AutoInferencePlugin : public IE::InferencePluginInternal {
 
         return impl;
     }
+
+private:
+    static thread_local ConfigType _autoConfig;
```

Comment:
why?

Reply:
As we want to use the configuration during LoadNetwork, so we will need to cache the config. but maybe we will update the _config in different thread, so I think each thread should has its own config, am I thinking too far away?

Reply:
just guard the config update by a mutex, as in MULTI afaik

Reply:
we could guard the config update by mutex like MULTI, but `GetOptimizationCapabilities()` still cannot get the correct config, what I mean is:
thread 1: loadNetwork(): update config --> GetOptimizationCapabilities(): read config
thread 2: loadNetwork(): update config --> GetOptimizationCapabilities(): read config
thread 3: loadNetwork(): update config --> GetOptimizationCapabilities(): read config
even we could lock update/read config, we couldnot make sure the GetOptimizationCapabilities() in thread1 read the config by thread1.

Reply:
Why do we even need to care about the issue if someone changes state from multiple threads?

Reply:
so just to make sure again, do you really think we could ignore the update and read config thread safe and alignment issue? if so, I will remove the handling for mulit-thread, basically as below:
    LoadNetwork() --> save a config copy to a member --> Get auto capabilites() -> read the copied config.

Reply:
updated, only lock the update config, thanks!